sympy/polys/rings.py

```diff
@@ -2532,6 +2566,135 @@ def compose(f, x, a=None):
 
         return poly
 
+    def primitive_wrt(self, x):
+        """
+        Returns the content and primitive part of a polynomial with respect to a
+        specified variable.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> R, x = ring("x", ZZ)
+        >>> p = 6*x**2 + 9*x**3
+        >>> p.primitive_wrt(x)
+        (9*x**3 + 6*x**2, 1)
+
+        """
+        p = self
+        coeffs = p.coeff_split({x})
+        cont = gcd_extract(coeffs)
+        prim = p.exquo(cont)
+        return cont, prim
+
+    def coeff_split_syms(self, syms):
+        """
+        Split a polynomial into two parts: terms involving specified symbolic
+        variables and terms without them.
+
+        This function divides the polynomial into terms that involve the
+        specified symbolic variables (syms) and terms that do not. It organizes
+        these terms into separate dictionaries.
```

Comment:
This explanation does not capture what the function does and actually is just incorrect.

What this method does is to treat a polynomial over several variables as if it were a polynomial over a subset of the variables whose coefficients are polynomials over the remaining variables e.g. we treat a polynomial in `K[x,y,z,t]` as if it were a polynomial in `K[x,t][y,z]`.

Reply:
Is there any significant benefit in having this method at all rather than just having `coeff_split`? This could just be a private method, like `_coeff_split_syms` and then it does not need a big docstring.